diff --git a/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.cpp b/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.cpp
index 779e428..6509900 100644
--- a/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.cpp
+++ b/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.cpp
@@ -26,7 +26,7 @@
 #define GxGDEM0213B74_PU_DELAY 300
 
 GxGDEM0213B74::GxGDEM0213B74(GxIO& io, int8_t rst, int8_t busy) :
-  GxEPD(GxGDEM0213B74_WIDTH, GxGDEM0213B74_HEIGHT), IO(io),
+  GxEPD(GxGDEM0213B74_VISIBLE_WIDTH, GxGDEM0213B74_HEIGHT), IO(io),
   _current_page(-1), _using_partial_mode(false), _diag_enabled(false), _power_is_on(false),
   _rst(rst), _busy(busy)
 {
@@ -41,10 +41,10 @@ void GxGDEM0213B74::drawPixel(int16_t x, int16_t y, uint16_t color)
   {
     case 1:
       swap(x, y);
-      x = GxGDEM0213B74_WIDTH - x - 1;
+      x = GxGDEM0213B74_VISIBLE_WIDTH- x - 1;
       break;
     case 2:
-      x = GxGDEM0213B74_WIDTH - x - 1;
+      x = GxGDEM0213B74_VISIBLE_WIDTH- x - 1;
       y = GxGDEM0213B74_HEIGHT - y - 1;
       break;
     case 3:
diff --git a/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.h b/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.h
index 62f0beb..3e43303 100644
--- a/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.h
+++ b/Main/libraries/GxEPD-3.1.3/src/GxGDEM0213B74/GxGDEM0213B74.h
@@ -25,6 +25,9 @@
 #define GxGDEM0213B74_WIDTH GxGDEM0213B74_X_PIXELS
 #define GxGDEM0213B74_HEIGHT GxGDEM0213B74_Y_PIXELS
 
+// note: the visible number of display pixels is 122*250, see GDEH0213B72 V1.1 Specification.pdf
+#define GxGDEM0213B74_VISIBLE_WIDTH 122
+
 #define GxGDEM0213B74_BUFFER_SIZE (uint32_t(GxGDEM0213B74_WIDTH) * uint32_t(GxGDEM0213B74_HEIGHT) / 8)
 
 // divisor for AVR, should be factor of GxGDEM0213B74_HEIGHT
